sympy/core/symbol.py

```diff
@@ -737,3 +739,53 @@ def traverse(symbols, frame):
         del frame  # break cyclic dependencies as stated in inspect docs
 
     return syms
+
+def disambiguate(*iter):
+    """
+    Return a Tuple containing the passed expressions with distinct
+    symbols(that appear the same when printed) replaced with numerically
```

Comment:
space before "(" and it's worth noting that all free symbols that were Dummies will now be Symbols.

Reply:
> it's worth noting that all free symbols that were Dummies will now be Symbols.

Yes, I noticed it earlier and thought that rather than converting everything to `Symbol`, the conversion should be according to the class they belong. So I tried something like this:
`reps[mapping[k][i]] = mapping[k][i].__class__("%s_%i"%(k, i), **mapping[k][i].assumptions0)`, but  this created a slight problem in creating tests for `Dummy` items(two identical dummies are not equal), therefore I thought it would be better if everything is `Symbol`.